spec/services/finance/npq/course_statement_calculator_spec.rb

```diff
@@ -246,4 +250,165 @@
       end
     end
   end
+
+  describe "#course_has_targeted_delivery_funding?" do
+    let(:cohort) { create(:cohort, start_year: 2022) }
+    let(:statement) { create(:npq_statement) }
+
+    context "Early headship coaching offer" do
+      let!(:npq_course) { create(:npq_ehco_course) }
+
+      it do
+        expect(subject.course_has_targeted_delivery_funding?).to be false
+      end
+    end
+
+    context "Additional support offer" do
+      let!(:npq_course) { create(:npq_aso_course) }
+
+      it do
+        expect(subject.course_has_targeted_delivery_funding?).to be false
+      end
+    end
+
+    context "Leadership course" do
+      let!(:npq_course) { create(:npq_leadership_course) }
+
+      it do
+        expect(subject.course_has_targeted_delivery_funding?).to be true
+      end
+    end
+  end
+
+  describe "#targeted_delivery_funding_declarations_count" do
+    let(:cohort) { create(:cohort, start_year: 2022) }
+
+    let(:participant_profile) do
+      create(
+        :npq_application,
+        :accepted,
+        :eligible_for_funding,
+        npq_course:,
+        npq_lead_provider:,
+
+        eligible_for_funding: true,
+        targeted_delivery_funding_eligibility: true,
+      ).profile
+    end
+
+    context "when there are zero declarations" do
+      it do
+        expect(subject.targeted_delivery_funding_declarations_count).to be_zero
+      end
+    end
+
+    context "when there are targeted delivery funding declarations" do
+      before do
+        travel_to statement.deadline_date do
+          create(:npq_participant_declaration, :eligible, npq_course:, cpd_lead_provider:, participant_profile:)
+        end
+      end
+
+      it "is counted" do
+        expect(subject.targeted_delivery_funding_declarations_count).to eql(1)
+      end
+    end
+
+    context "when multiple declarations from same user of one type" do
+      let(:participant_declaration) do
+        create(
+          :npq_participant_declaration,
+          :eligible,
+          participant_profile:,
+          npq_course:,
+          cpd_lead_provider:,
+        )
+      end
+
+      before do
+        allow_any_instance_of(RecordDeclaration).to receive(:valid?).and_return(true)
```

Comment:
Maybe we can try and avoid any validations by being more specific on the declaration type? Happy to help to try to work around doing this